src/server/game/Entities/Player/Player.h

```diff
@@ -3133,8 +3133,8 @@ class TC_GAME_API Player : public Unit, public GridObject<Player>
         WorldLocation m_recall_location;
         uint32 m_recall_instanceId;
 
-        DeclinedName *m_declinedname;
-        Runes *m_runes;
+        std::shared_ptr<DeclinedName> m_declinedname;
```

Comment:
does shared_ptr fit that well here ? as that means it's ok that some object is storing the old m_declinedname object when reloading it, causing some nasty bugs hard to reproduce

Reply:
I was looking at the _LoadDeclinedNames logic which deletes the allocated memory during every load so I was convinced that a shared_ptr is fine here because the load method would just create a new ptr and orphan this one here which would then invoke the deallocation

Reply:
I believe the question was "why not unique_ptr?"

Reply:
ye, now rethinking about it, I think I went overboard with my logic. will fix